### ISSUE TITLE
ansible-lint crashes if ansible is present in the environment

### ISSUE THREAD

**Issue by nothub** [NixOS/nixpkgs#197392]
### Describe the bug
I am experiencing an issue when executing `ansible-lint` in a nix-shell. If the `ansible` package is also present in the environment, the application crashes with message: `ImportError: cannot import name 'main' from 'ansible.cli.config' (/nix/store/ga7vgfjikdk55mg3xqg68qn6ps4538nj-python3.9-ansible-core-2.12.5/lib/python3.9/site-packages/ansible/cli/config.py)`

### Steps To Reproduce
```bash
pkgs="nixpkgs=https://github.com/NixOS/nixpkgs/archive/c5203abb1329f7ea084c04acda330ca75d5b9fb5.tar.gz"

# this executes
nix-shell --pure --packages ansible-lint git -I ${pkgs} --run "ansible-lint playbook.yaml"

# this crashes
nix-shell --pure --packages ansible ansible-lint git -I ${pkgs} --run "ansible-lint playbook.yaml"
```

### Expected behavior
I expect the ansible-lint application to execute unaffected by the presence of another package in the environment.

### Log
```bash
WARNING: PATH altered to include /nix/store/x9na3pxf7134pq7dkn1kgy9df6lf1z4v-python3-3.9.13/bin
Traceback (most recent call last):
  File "/nix/store/385zg6szjfql36smhm1wladzjbl0jg8w-python3.9-ansible-core-2.13.0/bin/.ansible-config-wrapped", line 6, in <module>
    from ansible.cli.config import main
ImportError: cannot import name 'main' from 'ansible.cli.config' (/nix/store/ga7vgfjikdk55mg3xqg68qn6ps4538nj-python3.9-ansible-core-2.12.5/lib/python3.9/site-packages/ansible/cli/config.py)
Traceback (most recent call last):
  File "/nix/store/2qzmxfsn2g174lisjzf6ql50gbfwbms5-python3.9-ansible-lint-6.0.2/bin/.ansible-lint-wrapped", line 9, in <module>
    sys.exit(_run_cli_entrypoint())
  File "/nix/store/2qzmxfsn2g174lisjzf6ql50gbfwbms5-python3.9-ansible-lint-6.0.2/lib/python3.9/site-packages/ansiblelint/__main__.py", line 253, in _run_cli_entrypoint
    sys.exit(main(sys.argv))
  File "/nix/store/2qzmxfsn2g174lisjzf6ql50gbfwbms5-python3.9-ansible-lint-6.0.2/lib/python3.9/site-packages/ansiblelint/__main__.py", line 166, in main
    app = get_app()
  File "/nix/store/2qzmxfsn2g174lisjzf6ql50gbfwbms5-python3.9-ansible-lint-6.0.2/lib/python3.9/site-packages/ansiblelint/app.py", line 205, in get_app
    app = App(options=default_options)
  File "/nix/store/2qzmxfsn2g174lisjzf6ql50gbfwbms5-python3.9-ansible-lint-6.0.2/lib/python3.9/site-packages/ansiblelint/app.py", line 41, in __init__
    self.runtime = Runtime(isolated=True)
  File "/nix/store/7p7nkm1f8hp64381vhrghk0agqbw20r7-python3.9-ansible-compat-2.0.4/lib/python3.9/site-packages/ansible_compat/runtime.py", line 84, in __init__
    self.config = AnsibleConfig()
  File "/nix/store/7p7nkm1f8hp64381vhrghk0agqbw20r7-python3.9-ansible-compat-2.0.4/lib/python3.9/site-packages/ansible_compat/config.py", line 422, in __init__
    config_dump = subprocess.check_output(
  File "/nix/store/x9na3pxf7134pq7dkn1kgy9df6lf1z4v-python3-3.9.13/lib/python3.9/subprocess.py", line 424, in check_output
    return run(*popenargs, stdout=PIPE, timeout=timeout, check=True,
  File "/nix/store/x9na3pxf7134pq7dkn1kgy9df6lf1z4v-python3-3.9.13/lib/python3.9/subprocess.py", line 528, in run
    raise CalledProcessError(retcode, process.args,
subprocess.CalledProcessError: Command '['ansible-config', 'dump']' returned non-zero exit status 1.
```

### Additional context
I just started looking into nix so I have a really hard time telling if I am causing this error or if its another problem. I could not find help on other platforms, so I'm creating this issue.

### Notify maintainers
@sengaya
@SuperSandro2000 

### Metadata
```console
$ nix-shell -p nix-info --run "nix-info -m"
 - system: `"x86_64-linux"`
 - host os: `Linux 5.15.0-52-generic, Ubuntu, 20.04.5 LTS (Focal Fossa), nobuild`
 - multi-user?: `no`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.9.1`
 - channels(hub): `"nixpkgs-22.05"`
 - nixpkgs: `/home/hub/.nix-defexpr/channels/nixpkgs`
```

**Comment by SuperSandro2000** [comment]
This is probably because they use different ansible minor versions in the background. Probably both should be updated to the lastest version.

**Comment by lstrojny** [comment]
Maybe (likely!) this is the same issue as https://github.com/NixOS/nixpkgs/issues/197213

**Comment by nothub** [comment]
I just tested again and this was fixed at some point in time between c5203abb1329f7ea084c04acda330ca75d5b9fb5 and 545c7a31e5dedea4a6d372712a18e00ce097d462.
